findLongestString.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * This method is to find the longest element in a Array
+ *
+ * @since 5.0.0
+ * @category Array
+ * @param {Array} array The array to inspect.
+ * @returns {string} Returns  longest element in a Array
+ * @example
+ *
+ * findLongest(["abc", 'abcd', "abcde"])
+ * // => "abcde"
+ */
+function findLongest(array) {
+  if (!Array.isArray(array)) {
+    throw new TypeError('Expected an Array')
+  }
+  const otherTypeExit = array.every((item) => typeof item === 'string')
+  if (!otherTypeExit) {
+    throw new TypeError('input contains items that are not strings')
+  }
+  if (!array.length) {
+    return null
+  }
+  const arrayEveryItemLen = array.map((item) => item.length)
+  const maxLen = Math.max(...arrayEveryItemLen)
+  const longestStrArr = array.filter((item) => item.length === maxLen)
+  // if longest string has more than one return null
+  if (longestStrArr.length > 1) {
```

Comment:
Instead of returning null , can we return the first largest string in the arr. I think this will be give somthing as a output.

Reply:
and we can do this in single iteration by reduce as well.If you want i can share the snippet as well

Reply:
well, i want to see your snippets😆,it's good.

Reply:
well, i want to see your snippets😆,it's good.

Reply:
```javascript
  const longestStrArr = array.reduce((longest,cur) => cur.length > longest.length ? cur : longest,"");
```

Reply:
it's great, but i think it will not pass the test, the type of input must be Array, the type of element must be String

Reply:
actually this is just the replacement of lines 24 to 26 . and till line 24 we are sure that we have the array in input because in the starting of function we are checking for array.

Reply:
thanks，i will use it

Reply:
thanks，i will use it

Reply:
hello,i have committed !!